src/Network/Wai/Log/Options.hs

```diff
@@ -2,26 +2,38 @@ module Network.Wai.Log.Options (
 -- * Options & Timing
   Options(..)
 , ResponseTime(..)
+, logSendingResponse
 -- * Defaults
 , defaultOptions
 , defaultLogRequest
 , defaultLogResponse
 ) where
 
-import Data.Aeson.Types (Pair)
+import Data.Aeson.Types (Pair, Value)
+import Data.ByteString.Builder (Builder)
 import Data.String.Conversions (ConvertibleStrings, StrictText, cs)
 import Data.Text (Text)
 import Data.Time.Clock (NominalDiffTime)
+import Data.UUID (UUID)
 import Log
-import Network.HTTP.Types.Status
+import Network.HTTP.Types.Header (ResponseHeaders)
+import Network.HTTP.Types.Status (Status, statusCode, statusMessage)
 import Network.Wai
 
 -- | Logging options
 data Options = Options {
-    logLevel            :: LogLevel
-  , logRequest          :: Request -> [Pair]
-  , logSendingResponse  :: Bool
-  , logResponse         :: Request -> Response -> ResponseTime -> [Pair]
+    logLevel    :: LogLevel
+  , logRequest  :: UUID -> Request -> [Pair]
+  , logResponse :: UUID -> Request -> Response -> Value -> ResponseTime -> [Pair]
+  -- | An optional constructor of the response body log value.
+  --
+  -- Logging response body involves extracting it from @Response@ via IO operations,
+  -- therefore the function takes @Request@, @Status@ and @ResponseHeaders@
+  -- as arguments to decide whether the IO operations of body extraction have
+  -- to be permormed.
+  -- The resulting @Maybe@ function is the constructor of a loggable @Value@
+  -- from the body bytestring builder.
```

Comment:
As much as this Haddock is technically in the _correct_ place, it is a bit unreadable when rendered.

Consider putting it on `data Options` instead and mention the field `logBody`, or having a one line description here, and the rest on the upper-level Haddock.

Reply:
done.